TestingHelpers/MockFile.cs

```diff
@@ -525,12 +525,30 @@ public override IEnumerable<string> ReadLines(string path, Encoding encoding)
 #if NET40
         public override void Replace(string sourceFileName, string destinationFileName, string destinationBackupFileName)
         {
-            throw new NotImplementedException(StringResources.Manager.GetString("NOT_IMPLEMENTED_EXCEPTION"));
+            Replace(sourceFileName, destinationFileName, destinationBackupFileName, false);
         }
 
         public override void Replace(string sourceFileName, string destinationFileName, string destinationBackupFileName, bool ignoreMetadataErrors)
         {
-            throw new NotImplementedException(StringResources.Manager.GetString("NOT_IMPLEMENTED_EXCEPTION"));
+            if (!mockFileDataAccessor.FileExists(sourceFileName))
+            {
+                throw new FileNotFoundException(string.Format(CultureInfo.InvariantCulture, StringResources.Manager.GetString("COULD_NOT_FIND_FILE_EXCEPTION"), sourceFileName));
+            }
+
+            if (!mockFileDataAccessor.FileExists(destinationFileName))
+            {
+                throw new FileNotFoundException(string.Format(CultureInfo.InvariantCulture, StringResources.Manager.GetString("COULD_NOT_FIND_FILE_EXCEPTION"), destinationFileName));
+            }
+
+            var mockFile = new MockFile(mockFileDataAccessor);
+
+            if (destinationBackupFileName != null)
```

Comment:
Sanity check -- what happens when the destinationBackupFileName doesn't exist in System.IO? i.e. The value is not null, but the directory doesn't exist. Or the backupfile already exists.

I think this would throw a DirectoryNotFoundException, which isn't listed as a possibility in the documentation. Might be a good candidate for a test.

Reply:
Testing of `System.IO.File.Replace` showed that it didn't care if backup file already existed, it would just override it. But if the directory that the backup file path is in didn't exist, there would be a `DirectoryNotFoundException`. Added tests to assert `MockFile` replicates this behavior.

Reply:
Alright cool, thats what I figured as its been a common theme with the other File create methods. They usually throw a Directory error if one doesn't exist, but I didn't see it in the documentation. Must be a gap on their side.